mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -80,7 +96,18 @@ class CrossValidator(override val uid: String) extends Estimator[CrossValidatorM
     val epm = $(estimatorParamMaps)
     val numModels = epm.length
     val metrics = new Array[Double](epm.length)
-    val splits = MLUtils.kFold(dataset.rdd, $(numFolds), 0)
+    val splits = if (schema.fieldNames.contains($(stratifiedCol))) {
```

Comment:
Agree. This could perform StratifiedSampling on a column "None" without the user knowing. Added a check to see if user has explicitly set the column.